Temperature conversion

Mustafa-Mangal opened the pull request:
---
name: solution review 
about: A template PR for code review with a checklist
---

<!--
  make this PR easy to find:

  - assign: yourself
  - link it to the corresponding challenge issue on the project board
-->

<!-- describe the challenge you will solve -->

## Behavior

### Files

- [x] The file name describes the function's behavior
- [x] There is a module docstring in the function file
- [x] The test file's name matches the function file name -
  `/tests/test_file_name.py`
- [x] There is a module docstring in the tests file

### Unit Tests

- [x] The test class has a helpful name in PascalCase
- [x] The test class has a docstring
- Every unit test has
  - [x] A helpful name
  - [x] A clear docstring
  - [x] Only one assertion
  - [x] There is no logic in the unit test
- [x] All tests pass
- [x] There are tests for defensive assertions
- [x] There are tests for boundary cases

### Function Docstring

- [x] The function's behavior is described
- The function's arguments are described:
  - [x] Type
  - [x] Purpose
  - [x] Other assumptions (eg. if it's a number, what's the expected range?)
- The return value is described
  - [x] Type
  - [x] Other assumptions are documented
- The defensive assertions are documented using `Raises:`
  - [x] Each assumption about an argument is checked with an assertion
  - [x] Each assertion checks for _only one_ assumption about the argument
- [x] Include 3 or more (passing!) doctests

### The Function

- [x] The function's name describes it's behavior
- [x] The function's name matches the file name
- [x] The function has correct type annotations
- [x] The function is not called in the function file

## Strategy

### Do's

- [x] Variable names help to understand the strategy
- [x] Any comments are clear and describe the strategy
- [x] Lines of code are spaced to help show different stages of the strategy

### Don'ts

- [x] The function's strategy _is not_ described in the documentation
- [x] Comments explain the _strategy_, **not** the _implementation_
- [x] The function _does not_ have more comments than code
  - If it does, consider finding a new strategy or a simpler implementation

## Implementation

- [x] The code passes the formatting checks
- [x] The code passes all Ruff linting checks
- [x] The code has no (reasonable) Pylint errors
  - In code review, you can decide when fixing a Pylint error is helpful and
    when it's too restricting.
- [x] Variables are named with snake_case
- [x] Variable names are clear and helpful
- [x] The code follows the strategy as simply as possible
- [x] The implementation is as simple as possible given the strategy
- [x] There are no commented lines of code
- [x] There are no `print` statements anywhere
- [x] The code includes defensive assertions
- [x] Defensive assertions include as little logic as possible

Mustafa-Mangal committed:
Temperature conversion

Mustafa-Mangal committed:
reformated

bekuretsion commented:
Dear Mustafa,It is good Job and keep it doing amazing stuff.I just have some suggestions

1.Module Docstrings(Function File (temperature_conversion.py)):It could be slightly expanded to describe any assumptions or edge cases (like handling of non-numeric values), but it’s acceptable((e.g., celsius should be a number) )
2.Defensive Assertions (Raises):
Although the function itself does not currently handle defensive assertions directly (e.g., ensuring celsius is a number), the tests appropriately handle this. If defensive assertions were to be included in the function, they would be better placed here to raise an exception for non-numeric inputs but it is good also. Example if not isinstance(celsius, (int, float)):
    raise TypeError("Input must be a number.")
3.Type Annotations: The function has no type annotations. Adding type annotations for the argument and return type would improve clarity and readbility and it would be a good practice as well.Example def celsius_to_fahrenheit(celsius: float) -> float:...

Mustafa-Mangal committed:
CI checks updates

Mustafa-Mangal committed:
Group repo readme file is ready

Mustafa-Mangal committed:
Remove unwanted files from temperature_conversion_function branch

Mustafa-Mangal committed:
- Expanded module docstring with assumptions and edge cases
- Added defensive assertions for numeric input validation
- Introduced type annotations for clarity and robustness

Mustafa-Mangal commented:
> Dear Mustafa,It is good Job and keep it doing amazing stuff.I just have some suggestions
> 
> 1.Module Docstrings(Function File (temperature_conversion.py)):It could be slightly expanded to describe any assumptions or edge cases (like handling of non-numeric values), but it’s acceptable((e.g., celsius should be a number) ) 2.Defensive Assertions (Raises): Although the function itself does not currently handle defensive assertions directly (e.g., ensuring celsius is a number), the tests appropriately handle this. If defensive assertions were to be included in the function, they would be better placed here to raise an exception for non-numeric inputs but it is good also. Example if not isinstance(celsius, (int, float)): raise TypeError("Input must be a number.") 3.Type Annotations: The function has no type annotations. Adding type annotations for the argument and return type would improve clarity and readbility and it would be a good practice as well.Example def celsius_to_fahrenheit(celsius: float) -> float:...


#This changes have been added

Mustafa-Mangal committed:
Restore deleted file: solutions/simple_interest_calculator.py

Mustafa-Mangal committed:
Remove simple_interest_calculator.py from temperature_conversion_function branch

bekuretsion reviewed (commented):
Good Job

bekuretsion reviewed (approved):
There is a kind of problem,why the simple calculator existed here..

Mustafa-Mangal commented:
> There is a kind of problem,why the simple calculator existed here..

There was an issue where the simple_interest_calculator.py file was unintentionally merged into the new branch. This happened because the Python file checks were associated with the Simple Interest Calculator, and it was accidentally included in the merge. I have since deleted the file, but it still appears in the commit history.
